pkg/streamingpromql/optimize/ast/prune_toggles.go

```diff
@@ -16,13 +16,29 @@ import (
 // e.g. `(avg(rate(foo[1m]))) and on() (vector(1) == 1)` -> `(avg(rate(foo[1m])))`
 // e.g. `(avg(rate(foo[1m]))) and on() (vector(1) == -1)` -> `(vector(1) == -1)`
 type PruneToggles struct {
-	mapper astmapper.ASTMapper
+	mapper *ASTExprMapperWithState
+
+	pruneTogglesMetrics
+}
+
+type pruneTogglesMetrics struct {
+	pruneTogglesAttempts  prometheus.Counter
+	pruneTogglesSuccesses prometheus.Counter
 }
 
-func NewPruneToggles() *PruneToggles {
-	mapper := astmapper.NewASTExprMapper(&pruneToggles{})
+func NewPruneToggles(reg prometheus.Registerer) *PruneToggles {
+	metrics := pruneTogglesMetrics{
+		pruneTogglesAttempts: promauto.With(reg).NewCounter(prometheus.CounterOpts{
+			Name: "cortex_mimir_query_engine_prune_toggles_attempted_total",
+			Help: "Total number of queries that the optimization pass has attempted to rewrite by pruning toggles.",
+		}),
+		pruneTogglesSuccesses: promauto.With(reg).NewCounter(prometheus.CounterOpts{
+			Name: "cortex_mimir_query_engine_prune_toggles_succeeded_total",
+			Help: "Total number of queries where the optimization pass has successfully rewritten by pruning toggles.",
+		}),
```

Comment:
The term "success" here might be a bit misleading - doesn't this count the number of times the expression was changed? ("success" to me implies it didn't fail with an error, but doesn't mean that it changed the expression)

Perhaps "rewritten expressions" or "applied expressions" would be less ambiguous?

Reply:
Using "rewritten" instead.